packages/signers/src/transaction-partial-signer.ts

```diff
@@ -0,0 +1,29 @@
+import { Address } from '@solana/addresses';
+import { CompilableTransaction } from '@solana/transactions';
+
+import { SignatureDictionary } from './types';
+
+/** Defines a signer capable of signing transactions. */
+export type TransactionPartialSigner<TAddress extends string = string> = Readonly<{
+    address: Address<TAddress>;
+    signTransaction(transactions: readonly CompilableTransaction[]): Promise<readonly SignatureDictionary[]>;
```

Comment:
nit: Same nit here on plurality.

Reply:
Moving this conversation [here](https://github.com/solana-labs/solana-web3.js/pull/1710#discussion_r1389591624).